sklearn/preprocessing/_data.py

```diff
@@ -103,7 +104,8 @@ def _handle_zeros_in_scale(scale, copy=True, constant_mask=None):
         if scale == 0.0:
             scale = 1.0
         return scale
-    elif isinstance(scale, np.ndarray):
+    else:
+    #elif isinstance(scale, np.ndarray):  # TODO instead of checking for array, check for array API compliance, how?
```

Comment:
This clearly needs to be fixed: how to check whether scale is compatible to the array API?

Reply:
I think we can keep the same branching logic unchanged.

[xp.finfo](https://data-apis.org/array-api/latest/API_specification/generated/array_api.finfo.html) should still be usable to compute the constant mask.

Reply:
As far as I know the only safe way to make a copy of an array is to use `xp.asarray` as follows:

```python
    scale = xp.asarray(scale, copy=True)
```